internal/support/logging/init.go

```diff
@@ -133,7 +133,7 @@ func connectToConsul(conf *ConfigurationStruct) (*ConfigurationStruct, error) {
 		}
 		conf = actual
 		//Check that information was successfully read from Consul
-		if len(conf.Persistence) == 0 {
```

Comment:
I'm trying to understand how / why the need for this logical check (i.e. `conf.ResendLimit == 0`) was obviated instead of perhaps 'rolling' this logical check into (or alongside) the one for `conf.Service.Port == 0`.

Reply:
The linked bug shows that there is a race in using Consul with services where if a service was started *immediately* at the same time as consul, consul would be up and return garbage data to the service, resulting in the service effectively having an uninitialized struct. This check ensures that the struct was in fact initialized with some sensible data. A program is never allowed to explicitly bind to port 0 (well you can make the syscall and this lets the OS choose a different port other than 0 for you) so if the port is 0 then we know that the struct is uninitialized and we need to wait for consul longer.

Reply:
Make sense, cool 💯